openhtf/plugs/acroname/power_module_plug.py

```diff
@@ -0,0 +1,306 @@
+# limitations under the License.
+
+
+"""Acroname power module plug for OpenHTF."""
+
+
+import logging
+import time
+from time import sleep
+
+import openhtf.conf as conf
+import openhtf.plugs as plugs
+
+from brainstem import discover
+from brainstem.link import Spec
+from brainstem.stem import MTMPM1
+from brainstem.stem import MTMEtherStem
+from brainstem.stem import USBStem
+from brainstem.defs import model_info
+
+# _LOG = logging.getLogger("openhtf.example.plugs.acroname.power_module_plug.py")
+_LOG = logging.getLogger(__name__)
+
+conf.Declare('serial_number_of_EtherStem',
+             description='Serial number of EtherStem that connects PowerModule to Ethernet.')
+
+conf.Declare('default_power_module_voltage_output',
+             description='Defalut power module voltage output in uV.')
+
+conf.Declare('default_power_module_current_limit',
+             description='Defalut power module current limit in uA.')
+
+class NoneEtherStemFoundError(Exception):
+ """No EtherStem module is found on the network."""
+
+class NoMatchingEtherStemFoundError(Exception):
+ """No EtherStem module matching the target EtherStem is found on the network."""
+
+class ConnectingToEtherStemError(Exception):
+ """Error connecting to EtherStem."""
+
+class ConnectingToPowerModuleThroughEtherStemError(Exception):
+ """Error connecting power module through EtherStem."""
+
+class ConfigurePowerSupplyError(Exception):
+ """Error configuring power supply."""
+
+class TurnOnPowerSupplyError(Exception):
+ """Error turning on power supply."""
+
+class ChangeVoltageError(Exception):
+ """Error changing voltage."""
+
+class ChangeCurrentLimitError(Exception):
+ """Error changing current limit."""
+
+class TurnOffPowerSupplyError(Exception):
+ """Error turning off power supply."""
+
+ 
+
+
+class PowerSupplyControl(plugs.BasePlug):   # pylint: disable=no-init
+  """Acronmae power module control plug."""
+  
+
+  def __init__(self):
+    config = conf.Config()
+    self.serial_number = config.serial_number_of_EtherStem
+    self.voltage = config.default_power_module_voltage_output
+    self.current_limit = config.default_power_module_current_limit
+    self.MTMPowerModuleAddress = 6
+    self.routerAddress = 4
+    self.EtherStem = MTMEtherStem()
+    self.power_module = MTMPM1()
+
+  def DiscoverAndConnectModule(self):
+    """
+    Discover all EtherStems on network.
+    Connect the one with the serial number specified in config yaml file.
+    The connection from server to EtherStem is Ethernet via TCPIP.
+    """
+    # codes below is to discover EtherStem on network
+    stem_list = discover.find_all(Spec.TCPIP)
+    EtherStemFound = False
+    if stem_list is not []:
+      for stem in stem_list:
+        print "stem: "+str(stem)
+        if stem.serial_number == self.serial_number:
+          stem_index = stem_list.index(stem)
+          spec = stem_list[stem_index]
+          EtherStemFound = True
+          break
+      if not EtherStemFound:
```

Comment:
The else of a for-loop is executed when the loop was not broken out of. Remove the EtherStemFound flag variable in favor of this:
```
for stem in stem_list:
  if ...:
    break
else:
  _LOG...
  raise ...
```